mobile/src/reducers/people/people.reducer.ts

```diff
@@ -6,107 +6,63 @@ import { ActionsObservable } from 'redux-observable';
 
 import { User } from '../user/user.model';
 import { Employee } from '../organization/employee.model';
-import { OrganizationActions, loadEmployeesForDepartment, loadEmployeesForRoom, LoadDepartmentsFinished, LoadEmployeeFinished } from '../organization/organization.action';
-import { UserActions, LoadUserEmployeeFinished } from '../user/user.action';
+import { OrganizationActions, loadEmployeesForDepartment, loadEmployeesForRoom, LoadDepartmentsFinished } from '../organization/organization.action';
 import { PeopleActions } from './people.action';
-import { EmployeeMap } from '../organization/employees.reducer';
 import { Department } from '../organization/department.model';
-import { DepartmentsTree } from '../../people/departments/departments-tree';
-import { DepartmentsTreeNode, stubIdForSubordinates } from '../../people/departments/departments-tree-node';
+import { LoadUserEmployeeFinished } from '../user/user.action';
 
 export interface PeopleState {
-    headDepartment: Department;
     departments: Department[];
-    departmentsHeads: Employee[];
-    employees: Employee[];
-    departmentsHeadsIds: string[];
-    departmentsBranch: DepartmentsTreeNode[];
-    departmentsTree: DepartmentsTree;
+    departmentsBranch: Department[];
+    currentFocusedDepartmentId: string;
 }
 
 const initState: PeopleState = {
-    headDepartment: null,
     departments: [],
-    departmentsHeads: [],
-    employees: [],
-    departmentsHeadsIds: [],
     departmentsBranch: [],
-    departmentsTree: null
+    currentFocusedDepartmentId: null
 };
 
-function shortDepartmentsTreeFor(departments: Department[]) {
-    const topLevelDepartment: Department = departments.find((department) => department.isHeadDepartment === true);
-    const departmentsTree: DepartmentsTree = {
-        root: {
-            departmentId: topLevelDepartment.departmentId,
-            departmentAbbreviation: topLevelDepartment.abbreviation,
-            departmentChiefId: topLevelDepartment.chiefId,
-            parent: null,
-            children: shortChildrenNodes(topLevelDepartment, departments)
-        }
-    };
-
-    return departmentsTree;
-}
-
 function onlyUnique(value: string, index: number, self: string[]) { 
     return self.indexOf(value) === index;
 }
 
-function shortChildrenNodes(headDeaprtment: Department, departments: Department[]) {
-    const nodes: DepartmentsTreeNode[] = [];
-    const siblings = departments.filter((subDepartment) => subDepartment.parentDepartmentId === headDeaprtment.departmentId);
+function departmentsBranchFromDepartmentWithId(departmentId: string, departments: Department[]) {
```

Comment:
what's the purpose of that? If you tell me what it's supposed to do I think we can make it much easier

Reply:
build departments set from up to bottom for a some department what should be focused on - user's or an employee (from profile's organization button).

It goes up till the head and goes down till the root. as result we have correct sequence of departments in focus for all levels.